include/ofi.h

```diff
@@ -266,8 +266,10 @@ static inline void *ofi_get_page_end(const void *addr, size_t page_size)
 static inline size_t
 ofi_get_page_bytes(const void *addr, size_t len, size_t page_size)
 {
-	return (char *)ofi_get_page_end((const char *) addr + len, page_size) -
-	       (char *)ofi_get_page_start(addr, page_size);
+	char *start = ofi_get_page_start(addr, page_size);
+	char *end = (char *)ofi_get_page_start((const char*)addr + len - 1, page_size)
+		    + page_size;
+	return end - start;
```

Comment:
Don't we just need to add 1 to the original return value?  We could also add an assert that the return value mod page_size is 0.

Reply:
Add 1 will cause extra page to added to the size, when the input are page aligned.

When ```addr``` and ```len``` are both aligned,  using the old implementation  ```ofi_get_page_end((const char *) addr + len, page_size)``` would be ```addr + len + page_size - 1```,  ```ofi_get_page_start(addr, page_size)``` would be addr. therefore the output of old implementation is ```len + page_size - 1```, plus 1 would be ```len + page```, while we want ```len``` for this case.

Adding assertion is a good idea.

Reply:
Assertion added in new revision.

Reply:
The first valid address is 'addr', and the last valid address is 'addr + len - 1'.  So, I think the problem with the original code is we want page_end(addr + len - 1) + 1.  That's basically what you have, only you used page_start(), then added page_size to get to the same location.